test/units/module_utils/common/collections.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+"""Test low-level utility functions from ``module_utils.common.collections``."""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from collections.abc import Sequence
+
+import pytest
+
+from ansible.module_utils.common.collections import is_iterable, is_sequence
+
+
+class SeqStub:
+    """Stub emulating a sequence type.
+
+    >>> from collections.abc import Sequence
+    >>> assert issubclass(SeqStub, Sequence)
+    >>> assert isinstance(SeqStub(), Sequence)
+    """
+
+
+Sequence.register(SeqStub)
+
+
+class IteratorStub:
+    def __next__(self):
+        raise StopIteration
+
+
+class IterableStub:
+    def __iter__(self):
+        return IteratorStub()
+
+
+TEST_STRINGS = 'he', b'he', u'Україна', u'Česká republika'
+
+TEST_ITEMS_SEQUENCES_NEGATIVE = (
+    {}, object(), frozenset(),
+    4, 0.,
+) + TEST_STRINGS
+
+TEST_ITEMS_SEQUENCES = (
+    [], (),
+    SeqStub(),
+)
+TEST_ITEMS_SEQUENCES = TEST_ITEMS_SEQUENCES + (
+    # Iterable effectively containing nested random data:
+    TEST_ITEMS_SEQUENCES_NEGATIVE,
+)
+
+
+@pytest.mark.parametrize('sequence_input', TEST_ITEMS_SEQUENCES)
+def test_sequence_positive(sequence_input):
+    """Test that non-string item sequences are identified correctly."""
+    assert is_sequence(sequence_input)
+    assert is_sequence(sequence_input, include_strings=False)
+
+
+@pytest.mark.parametrize('non_sequence_input', TEST_ITEMS_SEQUENCES_NEGATIVE)
+def test_sequence_negative(non_sequence_input):
+    """Test that non-sequences are identified correctly."""
+    assert not is_sequence(non_sequence_input)
+
+
+@pytest.mark.parametrize('string_input', TEST_STRINGS)
+def test_sequence_string_types(string_input):
+    """Test that ``is_sequence`` can separate string and non-string."""
+    assert is_sequence(string_input, include_strings=True)
+    assert not is_sequence(string_input, include_strings=False)
```

Comment:
Good unittests should have one call of the function being tested and one or more asserts about what happened when you called that function.  So this test should be split into two.